src/utils/decomposed.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { JsonMap, getString } from '@salesforce/ts-types';
+import { MetadataType } from '../registry/types';
+/** handle wide-open reading of values from elements inside any metadata xml file...we don't know the type
+ * Return the value of the matching element if supplied, or defaults `fullName` then `name`  */
+export const extractUniqueElementValue = (xml: JsonMap, uniqueId?: string): string | undefined =>
```

Comment:
when CustomObjects were the only decomposed thing, we relied on some default behaviors (xml names match filenames, directories are the xml elements, there's always a fullName) but that's not true across other types that we will support decomposing